### ISSUE TITLE
PostgresAdapter::getColumns does not handle postgis columns well

### ISSUE THREAD

**Issue by MasterOdin** [cakephp/phinx#1674]
The `getColumns` function for the PostgresAdapter does not handle geometric or geographic types very well, missing in capturing the defined shape, as well as set coordinate system. This is due to the fact that this information is not set in the `information_schema.columns` table, but rather Postgis defines two secondary tables `geometry_columns` and `geography_columns` tables. While cycling through each column, if the type is `geography` or `geometry`, query the above tables as appropriate (e.g. `SELECT * FROM public.geography_columns WHERE f_table_schema = %s AND f_table_name = %s`).

**Comment by dereuromark** [comment]
I this still open?

**Comment by MasterOdin** [comment]
Yes, though I will probably make a new issue that combines this one and #1506, as the geometry handling in phinx needs quite a bit of work that should probably be done all at the same time. Leaving this open until such an issue is made though.
